eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/sink/handle/CommonHttpSinkHandler.java

```diff
@@ -0,0 +1,197 @@
+        doInitWebClient();
+    }
+
+    private void doInitWebClient() {
+        final Vertx vertx = Vertx.vertx();
+        WebClientOptions options = new WebClientOptions()
+            .setDefaultHost(this.connectorConfig.getHost())
+            .setDefaultPort(this.connectorConfig.getPort())
+            .setSsl(this.connectorConfig.isSsl())
+            .setKeepAlive(this.connectorConfig.isKeepAlive())
+            .setKeepAliveTimeout(this.connectorConfig.getKeepAliveTimeout() / 1000)
+            .setIdleTimeout(this.connectorConfig.getIdleTimeout())
+            .setIdleTimeoutUnit(TimeUnit.MILLISECONDS)
+            .setConnectTimeout(this.connectorConfig.getConnectionTimeout())
+            .setMaxPoolSize(this.connectorConfig.getMaxConnectionPoolSize());
+        this.webClient = WebClient.create(vertx, options);
+    }
+
+
+    @Override
+    public void handle(ConnectRecord record) {
+        // create headers
+        MultiMap headers = HttpHeaders.headers()
+            .set(HttpHeaderNames.ACCEPT, "application/json; charset=utf-8");
+
+        // convert ConnectRecord to HttpConnectRecord
+        HttpConnectRecord httpConnectRecord = convertToHttpConnectRecord(record);
+
+        // send the request
+        this.webClient.post(this.connectorConfig.getPath())
+            .putHeaders(headers)
+            .sendJson(httpConnectRecord)
+            .onSuccess(res -> {
+                Long timestamp = record.getTimestamp();
+                Map<String, ?> offset = record.getPosition().getOffset().getOffset();
+                log.info("Request sent successfully. Record: timestamp={}, offset={}", timestamp, offset);
+                // Determine whether the status code is 200
+                if (res.statusCode() == HttpResponseStatus.OK.code()) {
+                    // store the received data, when webhook is enabled
+                    if (this.connectorConfig.isWebhook()) {
+                        String dataStr = res.body().toString();
+                        if (dataStr.isEmpty()) {
+                            log.warn("Received data is empty.");
+                            return;
+                        }
+                        JSONObject receivedData = JSON.parseObject(dataStr);
+                        if (receivedDataQueue.size() == Integer.MAX_VALUE) {
+                            // if the queue is full, remove the oldest element
+                            JSONObject removedData = receivedDataQueue.poll();
+                            log.info("The queue is full, remove the oldest element: {}", removedData);
+                        }
+                        boolean b = receivedDataQueue.offer(receivedData);
+                        if (b) {
+                            log.info("Successfully put the received data into the queue: {}", receivedData);
+                        } else {
+                            log.error("Failed to put the received data into the queue: {}", receivedData);
```

Comment:
IMHO, `b` is not a good naming~

Reply:
Kindly notice that I haven't seen any reply in this conversation.

Reply:
Modified but not yet submitted.

Reply:
Resolved at https://github.com/apache/eventmesh/pull/4837/commits/fe2b7327b744fcf0f3ea3576c892c551ed1d6fb4